### DISCUSSION TITLE
在patchRoutes函数中，getDvaApp()._store.dispatch一定不为null么

### DISCUSSION THREAD

**Discussion by zhangzhuang15** [umijs/umi#8634] [Q&A]
在app.ts进行动态配置时，我在patchRoutes函数中，使用getDvaApp()._store.dispatch拿到了dispatch, 多次刷新项目页面后，dispatch依旧可以拿到，而且都不是null，那我可不可以认为在patchRoutes函数里，执行getDvaApp()._store.dispatch就一定可以保证拿到的dispatch不是null呢？

**Comment by fz6m** [comment]
你拿到过 `null` 吗？

**Comment by zhangzhuang15** [comment]
你好 ，您发给我的邮件我已经收到。

**Comment by zhangzhuang15** [comment]
是的，我拿到过，更多情况是在render函数和patch Routes函数里，调用getDvaApp函数直接返回undefine了。



---原始邮件---
发件人: ***@***.***&gt;
发送时间: 2022年7月20日(周三) 晚上7:00
收件人: ***@***.***&gt;;
抄送: ***@***.******@***.***&gt;;
主题: Re: [umijs/umi] 在patchRoutes函数中，getDvaApp()._store.dispatch一定不为null么 (Discussion #8634)




 
你拿到过 null 吗？

**Reply by fz6m** [reply]
提供复现 demo ，提 issue 问吧。

**Reply by zhangzhuang15** [reply]
这个问题我今天上午看了umi源码找到答案了，还是要谢谢您的建议。
